Doc/c-api/init.rst

```diff
@@ -1279,6 +1279,15 @@ with sub-interpreters:
       Hangs the current thread, rather than terminating it, if called while the
       interpreter is finalizing.
 
+
+.. c:function:: int PyGILState_EnsureOrFail(PyGILState_STATE *state)
+
+   Similar to :c:func:`PyGILState_Ensure`, but *state* is an argument
+   and return ``-1`` if the thread must exit. Return ``0`` on success.
```

Comment:
This is confusing and misleading. The C or C++ thread must not exit. It's just that the function failed taking the GIL, probably because Python is finalized.